pkg/landscaper/installations/helper.go

```diff
@@ -322,3 +323,39 @@ func StringMapToRawMessageMap(m map[string]string) map[string]json.RawMessage {
 	}
 	return n
 }
+
+// HandleSubComponentPhaseChanges updates the phase of the given installation, if its phase doesn't match the combined phase of its subinstallations/executions anymore
+// it returns a boolean indicating whether the phase was changed
+func (o *Operation) HandleSubComponentPhaseChanges(ctx context.Context, logger logr.Logger) (bool, error) {
+	execRef := o.Inst.Info.Status.ExecutionReference
+	var exec *lsv1alpha1.Execution
+	if execRef != nil {
+		exec = &lsv1alpha1.Execution{}
+		err := o.Client().Get(ctx, execRef.NamespacedName(), exec)
+		if err != nil {
+			return false, fmt.Errorf("error getting execution for installation %s/%s: %w", o.Inst.Info.Namespace, o.Inst.Info.Name, err)
+		}
+	}
+	subinsts, err := o.ListSubinstallations(ctx)
+	if err != nil {
+		return false, fmt.Errorf("error fetching subinstallations for installation %s/%s: %w", o.Inst.Info.Namespace, o.Inst.Info.Name, err)
+	}
+	phases := []lsv1alpha1.ComponentInstallationPhase{}
+	if exec != nil {
+		phases = append(phases, lsv1alpha1.ComponentInstallationPhase(exec.Status.Phase))
+	}
+	for _, sub := range subinsts {
+		phases = append(phases, sub.Status.Phase)
+	}
+	cp := lsv1alpha1helper.CombinedInstallationPhase(phases...)
+	if o.Inst.Info.Status.Phase != cp {
```

Comment:
I think we also need to update the exports and notify dependent installations if the installation's status changes to succeeded